crates/bevy_core_pipeline/src/fxaa/fxaa.wgsl

```diff
@@ -0,0 +1,285 @@
+// NVIDIA FXAA 3.11
+// Original source code by TIMOTHY LOTTES
+// https://gist.github.com/kosua20/0c506b81b3812ac900048059d2383126
+//
+// Cleaned version - https://github.com/kosua20/Rendu/blob/master/resources/common/shaders/screens/fxaa.frag
+//
+// Tweaks by mrDIMAS - https://github.com/FyroxEngine/Fyrox/blob/master/src/renderer/shaders/fxaa_fs.glsl
+
+#import bevy_core_pipeline::fullscreen_vertex_shader
+
+@group(0) @binding(0)
+var screenTexture: texture_2d<f32>;
+@group(0) @binding(1)
+var samp: sampler;
+
+// Trims the algorithm from processing darks.
+#ifdef EDGE_THRESH_MIN_LOW
+    let EDGE_THRESHOLD_MIN: f32 = 0.0833;
+#endif
+
+#ifdef EDGE_THRESH_MIN_MEDIUM
+    let EDGE_THRESHOLD_MIN: f32 = 0.0625;
+#endif
+
+#ifdef EDGE_THRESH_MIN_HIGH
+    let EDGE_THRESHOLD_MIN: f32 = 0.0312;
+#endif
+
+#ifdef EDGE_THRESH_MIN_ULTRA
+    let EDGE_THRESHOLD_MIN: f32 = 0.0156;
+#endif
+
+// The minimum amount of local contrast required to apply algorithm.
+#ifdef EDGE_THRESH_LOW
+    let EDGE_THRESHOLD_MAX: f32 = 0.250;
+#endif
+
+#ifdef EDGE_THRESH_MEDIUM
+    let EDGE_THRESHOLD_MAX: f32 = 0.166;
+#endif
+
+#ifdef EDGE_THRESH_HIGH
+    let EDGE_THRESHOLD_MAX: f32 = 0.125;
+#endif
+
+#ifdef EDGE_THRESH_ULTRA
+    let EDGE_THRESHOLD_MAX: f32 = 0.063;
+#endif
+
+let ITERATIONS: i32 = 12; //default is 12
+let SUBPIXEL_QUALITY: f32 = 0.75;
+// #define QUALITY(q) ((q) < 5 ? 1.0 : ((q) > 5 ? ((q) < 10 ? 2.0 : ((q) < 11 ? 4.0 : 8.0)) : 1.5))
+fn QUALITY(q: i32) -> f32 {
+    switch (q) {
+        //case 0, 1, 2, 3, 4: { return 1.0; }
+        default:              { return 1.0; }
+        case 5:               { return 1.5; }
+        case 6, 7, 8, 9:      { return 2.0; }
+        case 10:              { return 4.0; }
+        case 11:              { return 8.0; }
+    }
+}
+
+fn rgb2luma(rgb: vec3<f32>) -> f32 {
+    return sqrt(dot(rgb, vec3<f32>(0.299, 0.587, 0.114)));
+}
+
+// https://gpuopen.com/learn/optimized-reversible-tonemapper-for-resolve/
+fn tonemap(c: vec3<f32>) -> vec3<f32> { 
```

Comment:
Should `tonemap` and `tonemap_invert` be renamed and moved somewhere else? We may also end up using these for dealing with HDR msaa in the future.